subhub/sub/payments.py

```diff
@@ -103,15 +103,14 @@ def _get_all_plans():
     return stripe_plans
 
 
-def check_stripe_subscriptions(customer: Customer) -> list:
+def retrieve_stripe_subscriptions(customer: Customer) -> list:
     try:
-        logger.debug("check stripe subscriptions", subscriptions=customer.subscriptions)
         customer_subscriptions_data = customer.subscriptions
         customer_subscriptions = customer_subscriptions_data.get("data")
         return customer_subscriptions
-    except NameError as ne:
-        logger.error("error getting subscriptions", customer=customer, error=ne)
-        return []
+    except AttributeError as e:
+        logger.error("error getting subscriptions", customer=customer, error=e)
+        raise AttributeError()
```

Comment:
why not raise the actual error instead of instantiating a new AttributeError?

Reply:
Good catch, updated.